lib/ipmi_fru.c

```diff
@@ -726,13 +737,15 @@ read_fru_area(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
 			break;
 		}
 		if (rsp->ccode) {
-			/* if we get C7h or C8h or CAh return code then we requested too
-			* many bytes at once so try again with smaller size */
-			if ((rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca)
-					&& fru->max_read_size > 8) {
-				if (fru->max_read_size > 32) {
-					/* subtract read length more aggressively */
-					fru->max_read_size -= 8;
+			/* if we get C7h or C8h or CAh return code then we
+			 * requested too many bytes at once so try again with
+			 * smaller size */
+			if (fru_cc_rq2big(rsp->ccode)
+			    && fru->max_read_size > FRU_BLOCK_SZ) {
```

Comment:
For multi-line conditions please put the opening curly brace on a separate line.